driver/sccb.c

```diff
@@ -205,6 +222,160 @@ int SCCB_Write16(uint8_t slv_addr, uint16_t reg, uint8_t data)
     return ret == ESP_OK ? 0 : -1;
 }
 
+uint8_t SCCB_Read16_Validate(uint8_t slv_addr, uint16_t reg) {
+  sccbLock();
+  ESP_LOGD(TAG, "SCCB_Read16: %x", reg);
+  uint8_t data       = 0;
+  esp_err_t ret      = ESP_FAIL;
+  uint16_t reg_htons = LITTLETOBIG(reg);
+  uint8_t* reg_u8    = (uint8_t*)&reg_htons;
+  int i;
+  for (i = 0; i < SCCB_RETRY_COUNT; i++) {
+    i2c_cmd_handle_t cmd = i2c_cmd_link_create();
+    i2c_master_start(cmd);
+    i2c_master_write_byte(cmd, (slv_addr << 1) | WRITE_BIT, ACK_CHECK_EN);
+    i2c_master_write_byte(cmd, reg_u8[0], ACK_CHECK_EN);
+    i2c_master_write_byte(cmd, reg_u8[1], ACK_CHECK_EN);
+    i2c_master_stop(cmd);
+    ret = i2c_master_cmd_begin(sccb_i2c_port, cmd, 1000 / portTICK_RATE_MS);
+    i2c_cmd_link_delete(cmd);
+    if (ret != ESP_OK) {
+      ESP_LOGE(TAG, "i2c_master_cmd_begin FAIL: %d", i);
+      continue;
```

Comment:
Are these `continue`s supposed to trigger the delay or just immediately retry?

Reply:
For simplicity, I moved the delay to the top of the for loop. The first pass through results in a vTaskDelay(0) anyway.